x-pack/plugins/security_solution/public/resolver/store/middleware/current_related_event_fetcher.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Dispatch, MiddlewareAPI } from 'redux';
+import { isEqual } from 'lodash';
+import { SafeResolverEvent } from '../../../../common/endpoint/types';
+
+import { ResolverState, DataAccessLayer, PanelViewAndParameters } from '../../types';
+import * as selectors from '../selectors';
+import { ResolverAction } from '../actions';
+
+/**
+ *
+ * @description - This api is called after every state change.
+ * If the current view is the `eventDetail` view it will request the event details from the server.
+ * @export
+ * @param {DataAccessLayer} dataAccessLayer
+ * @param {MiddlewareAPI<Dispatch<ResolverAction>, ResolverState>} api
+ * @returns {() => void}
+ */
+export function CurrentRelatedEventFetcher(
+  dataAccessLayer: DataAccessLayer,
+  api: MiddlewareAPI<Dispatch<ResolverAction>, ResolverState>
+): () => void {
+  let last: PanelViewAndParameters | undefined;
+
+  return async () => {
+    const state = api.getState();
+
+    const newParams = selectors.panelViewAndParameters(state);
+
+    const oldParams = last;
+    last = newParams;
+
+    // If the panel view params have changed and the current panel view is the `eventDetail`, then fetch the event details for that eventID.
+    if (!isEqual(newParams, oldParams) && newParams.panelView === 'eventDetail') {
+      const currentEventID = newParams.panelParameters.eventID;
+
+      api.dispatch({
+        type: 'appRequestedCurrentRelatedEventData',
+      });
+
+      let result: SafeResolverEvent | undefined | null;
+      try {
+        result = await dataAccessLayer.event(currentEventID);
+      } catch (error) {
+        api.dispatch({
+          type: 'serverFailedToReturnCurrentRelatedEventData',
+        });
+      }
+
+      if (result) {
+        api.dispatch({
+          type: 'serverReturnedCurrentRelatedEventData',
+          payload: result,
+        });
+      } else {
+        api.dispatch({
+          type: 'serverFailedToReturnCurrentRelatedEventData',
```

Comment:
❔ Just thinking out loud: is it different that maybe there wasn't an _error_ here, per se, but just that the server didn't _have_ a related event for that particular ID? Do we want to treat that differently in the UX because of maybe data sliding off the index or something? Like "Hey there, we may or may not have ever had an event like this, but we don't now". Especially if we're rehydrating from URLs that could maybe get shared and looked at 4-5 months down the line. Probably too big a question for this context, but I wanted to put it on the board.

Reply:
That's a good point. That would require a larger discussion because we wouldn't know whether or not it was an event that _could_ have been there or a random typo in the url or some other random thing. At the very least, for now though, we can convey that we couldn't find the data and then add potentially better reasoning around it in the future or just have additional info in the docs.

Reply:
Yeah, I'm thinking just on a basic level of should we show:
`Something broke` to be different than `We don't have that event`. That we "ever" had it is some kind of thing we probably can't say.

Reply:
Before releasing I think we should have 404-type messages as well as general error messages, or all panels.

If nothing was returned and the response code was successful we could show "your data was not found".

if the server returned an error status code we could say "there was an error"

Reply:
I like that idea. Maybe we use the `panel_content_error` component and repurpose that to something like `panel_content_status` where we pass it a status type `'loading' | 'error' | 'empty'` or similar and use it for all the panel states with a passed in message?